deb/debian-buster/Dockerfile

```diff
@@ -1,8 +1,8 @@
 ARG GO_IMAGE
-ARG CONTAINERD_SHIM_PROCESS_IMAGE
 ARG BUILD_IMAGE=debian:buster
+ARG ENGINE_IMAGE
 FROM ${GO_IMAGE} as golang
-FROM ${CONTAINERD_SHIM_PROCESS_IMAGE} as shim-process
+FROM ${ENGINE_IMAGE} as engine
```

Comment:
Lets come back to this in a follow up, but I think we're using `$ENGINE_IMAGE` inconsistently - in some places it's the full name with org, in others it's just the image name without the org prefix.